rpcs3/Emu/Cell/SPUThread.cpp

```diff
@@ -5469,12 +5482,71 @@ s64 spu_thread::get_ch_value(u32 ch)
 
 	case SPU_RdEventStat:
 	{
+		const bool is_spurs_task_wait = pc == 0x11a8 && group->max_run != 6 && spurs_addr;
```

Comment:
Can we move some of the thread scheduling to their own files/classes? It would greatly improve readability. It's something the emulator really needs in general.
In particular this case is so long it kinda breaks all expectations of what a switch-case should be. Just something to think about.